graphdatascience/query_runner/query_runner.py

```diff
@@ -7,9 +8,65 @@
 from .graph_constructor import GraphConstructor
 
 
+class EndpointType(Enum):
+    PROCEDURE = "PROCEDURE"
+    FUNCTION = "FUNCTION"
+
+
 class QueryRunner(ABC):
+    def call_function(
+        self,
+        endpoint: str,
+        yields: Optional[List[str]] = None,
+        body: Optional[str] = None,
+        params: Optional[Dict[str, Any]] = None,
+        database: Optional[str] = None,
+        custom_error: bool = True,
+    ) -> DataFrame:
+        return self.call_endpoint(
+            type=EndpointType.FUNCTION,
+            endpoint=endpoint,
+            yields=yields,
+            body=body,
+            params=params,
+            database=database,
+            custom_error=custom_error,
+        )
+
+    def call_procedure(
+        self,
+        endpoint: str,
+        yields: Optional[List[str]] = None,
+        body: Optional[str] = None,
+        params: Optional[Dict[str, Any]] = None,
+        database: Optional[str] = None,
+        custom_error: bool = True,
+    ) -> DataFrame:
+        return self.call_endpoint(
+            type=EndpointType.PROCEDURE,
+            endpoint=endpoint,
+            yields=yields,
+            body=body,
+            params=params,
+            database=database,
+            custom_error=custom_error,
+        )
+
+    @abstractmethod
+    def call_endpoint(
```

Comment:
It looks like this method is not really needed, if the only difference from the other `call_*` is that it takes a type parameter. And you could also remove the enum I think

Reply:
users of query_runner use call_proc/function but implementers only need call_endpoint. Renaming to `_call_endpoint` could make this clearer

Reply:
Sure, but if the enum is needed as an argument, it suggests that the different types need to be handled in different ways anyway

Reply:
short-cutted call_function to run_cypher as its always ended up there